src/loops.rs

```diff
@@ -300,6 +326,32 @@ impl<'v, 't> Visitor<'v> for VarVisitor<'v, 't> {
     }
 }
 
+fn var_used(expr: &Expr, def_id: Option<NodeId>, cx: &LateContext) -> bool {
+    match def_id {
+        None => false,
+        Some(def_id) => {
+            let mut visitor = VarUsedVisitor{ def_id: def_id, found: false, cx: cx };
+            walk_expr(&mut visitor, expr);
+            visitor.found
+        }
+    }
+}
+
+struct VarUsedVisitor<'v, 't: 'v> {
```

Comment:
This is quite a useful thing to have – I imagine we could use it in the shadows lint with a bit of tweaking.